strawberry/backends/base_backend.py

```diff
@@ -0,0 +1,33 @@
+from abc import ABC, abstractmethod
+from typing import Optional
+
+from strawberry.types import StrawberryObjectType
+
+
+class StrawberryBackend(ABC):
+
+    def __init__(self, query: StrawberryObjectType,
+                 mutation: Optional[StrawberryObjectType] = None,
+                 subscription: Optional[StrawberryObjectType] = None):
+        self.query = query
+        self.mutation = mutation
+        self.subscription = subscription
+
+    # TODO: Other parameters
+    @abstractmethod
+    def execute(self, query: str) -> ...:
+        # TODO: Rename query to document?
+        ...
+
+    # TODO: Other parameters
+    @abstractmethod
+    async def execute_async(self, query: str) -> ...:
+        ...
+
+    # TODO: Other parameters
+    @abstractmethod
+    async def subscribe(self, query: str) -> ...:
+        # TODO: Should you be able to make subscription queries from execute and
```

Comment:
no, subscription are only to be done here :)

Reply:
To me, it seems that if we ever want to _fully_ support the capabilities of GraphQL, we need to support creating a document with both queries and subscriptions at the same time. 

This could very well be a can to kick down the road, but it does make me think we'll need to create a more abstract function that handles all forms of queries/subscription/mutation documents. If we do this, I think the `execute`/`execute_sync`/`subscribe`/etc methods would need to become just convenience methods.